fix: pass `ManyToMany` `onUpdate` option to foreign key metadata

fan-tom opened the pull request:
Closes: #4980

I'm not sure if onDelete/onUpdate options are handled properly, it seems to me that they must be applied to foreign keys pointing to OPPOSITE entity, not to entity they are used in, but such changes break existing tests, so I considered to fix this issue not breaking them.

pleerock commented:
Based on your comments what I understood is that we need to set `onUpdate` and `onDelete` for every column in the junction table separately, am I right? If so, this temporary change doesn't make sense and we probably need to remove this option and introduce it for `@JoinTable`'s `joinColumn` option?:

```ts
@ManyToMany(
        () => Book,
        book => book.authors
    )
    @JoinTable({
        name: "author_to_books",
        joinColumn: {
            name: "author_id",
            referencedColumnName: "id",
            onDelete: "CASCADE",
            onUpdate: "CASCADE"
        },
        inverseJoinColumn: {
            name: "book_id",
            referencedColumnName: "id",
            onDelete: "CASCADE",
            onUpdate: "CASCADE"
        },
    })
``` 

Also, before any change will be merged we need to be sure nothing will be broken for exist users and ORM wont generate any new queries for those who update the version. I'm a bit scary to merge such kind of things.

fan-tom commented:
I just want to say that when I write
```typescript
class Author {
  @ManyToMany(type => Book, {
    onUpdate: 'CASCADE',
    onDelete: 'CASCADE',
  })
  books: Book[];
}

class Book {
  @JoinTable()
  @ManyToMany(type => Author, {
    onUpdate: 'NO ACTION',
    onDelete: 'NO ACTION',
  })
  authors: Author[];
}
```
I expect that I can update id's of books and get fk updated accordingly, but cannot update author's id's, like when you use ManyToOne and OneToOne, you apply those decorators inside referencing table and onDelete/onUpdate options are applied to links to REFERENCED table. But in case of ManyToMany when I set `onDelete: 'CASCADE'` inside `Author` entity, I get it applied to link to CURRENT table(Author), instead of link to REFERENCED table(Book).

What about affecting current users, I just added missing property handling code for `onUpdate` option, that works the same way `onDelete` works currently, just like people requested in issue, and fixed the bug that single `onDelete` option was used for both sides, so I don't understand what things you are scary about.

fan-tom commented:
Is it related issue? https://github.com/typeorm/typeorm/issues/3505

imnotjames reviewed (commented):
This makes sense.  Can you rebase and get the new test suite kicking off?

fabianscheidt commented:
I am facing both of the issues that are mentioned here. Wouldn't it make sense to merge @fan-tom's suggestion now and later come up with a change that involves the `@JoinTable` decorator as suggested by @pleerock?

Ami777 commented:
Why isn't this merged... 1 year later :|?

jcrben commented:
@fan-tom up for rebasing for a merge as suggested by @imnotjames?

fan-tom commented:
rebased

AlexMesser commented:
thank you for contribution!
